src/main/java/org/redcraft/redcraftprotect/listeners/inventoryListeners/InventoryMoveItemListener.java

```diff
@@ -0,0 +1,20 @@
+package org.redcraft.redcraftprotect.listeners.inventoryListeners;
+
+import org.bukkit.Bukkit;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.Listener;
+import org.bukkit.event.inventory.InventoryMoveItemEvent;
+import org.redcraft.redcraftprotect.RedCraftProtect;
+
+public class InventoryMoveItemListener implements Listener {
+
+    @EventHandler(priority = EventPriority.NORMAL)
+    public void onInventoryMoveItem(InventoryMoveItemEvent event) {
+
+        if(RedCraftProtect.getInstance().protectedElements.canBlockBeModified(event.getSource().getLocation(), event.getDestination().getLocation())){
```

Comment:
This should be cached on a memory level (bypass CacheManager) because it's very important to not hang the server if someone uses an auto clicker